external/external.sh

```diff
@@ -132,9 +136,29 @@ function parse_tag() {
 		*ubuntu*|*latest*|*nightly*) 
 	   		docker_os=ubuntu;;
    		*) echo "Unable to recognize DOCKER_OS from DOCKERIMAGE_TAG = $tag!";;
-	esac     
+	esac
+	
+}
+
+function parse_docker_args() {
+# parse docker_args to two variable: mountV and  imageArg
+	while [[ $# -gt 0 ]] && [[ ."$1" = .-* ]] ; do
+		opt="$1";
+		shift; 
+
+		case "$opt" in
+			"--volumn" | "-v" )
+				mountV="-v $1";
+				shift;;
+			"--image" | "-i" )
+				imageArg="$1"; 
+				shift;;
+			*) echo >&2 "Invalid docker args option: ${opt}"; exit 1;
```

Comment:
I am not sure that --volume and --image are the only 2 args that people use (EXTRA_DOCKER_ARGS mechanism was created to allow for multiple docker args from any of the [docker build options](https://docs.docker.com/engine/reference/commandline/build/#options) or [docker run options](https://docs.docker.com/engine/reference/commandline/run/#options) so not quite sure what happens if I try to run with more than those in my EXTRA_DOCKER_ARGS param, I guess other stuff gets thrown out).  

It may make sense to restrict (and wait to hear if anyone asks for more allowable options though it will reduce the flexibility one has in passing in arguments).  Fine for this PR, may want to consider it again later.

Reply:
Agreed that other args might need to use EXTRA_DOCKER_ARGS. For now -v and -i are used for different docker commands or different test steps( -v is used for `docker run` for running tests, and -i is used for `docker build` for building tests). To make it more flexible and support other needed args we can update to with -b ( all docker build related args) and -r (all docker run related args). I can open a issue in case we do need it.